drivers/volume/volume.go

```diff
@@ -751,3 +751,81 @@ func IsCSIDriverWithoutSnapshotSupport(pv *v1.PersistentVolume) bool {
 	}
 	return true
 }
+
+// GetNodeRegion - get the node region
+func GetNodeRegion() (string, error) {
+	nodes, err := core.Instance().GetNodes()
+	if err != nil {
+		logrus.Errorf("getNodeZones: getting node details failed: %v", err)
+		return "", fmt.Errorf("failed in getting the nodes: %v", err)
+	}
+	for _, node := range nodes.Items {
+		return node.Labels[v1.LabelTopologyRegion], nil
+	}
+	return "", fmt.Errorf("failed in getting the nodes: %v", err)
+}
+
+// GetNodeZones - get zone details of the nodes in the cluster.
+func GetNodeZones() ([]string, error) {
+	nodes, err := core.Instance().GetNodes()
+	if err != nil {
+		logrus.Errorf("getNodeZones: getting node details failed: %v", err)
+		return nil, fmt.Errorf("failed in getting the nodes: %v", err)
+	}
+	var nodeZoneList []string
+	for _, node := range nodes.Items {
+		nodeZone := strings.Split(node.Labels[v1.LabelTopologyZone], "-")
+		nodeZoneList = append(nodeZoneList, nodeZone[2])
```

Comment:
This should be a map instead of a list. Two nodes can be a part of the same zone.

Reply:
I see that you have a Unique() helper function. It would be better to filter out the list here itself in this function. so that callers of this function dont need to worry about calling Unique()

Reply:
Fixed. Adding the logic to filter it out in GetNodeZones itself.